appservice/src/tree/DeploymentsTreeItem.ts

```diff
@@ -92,10 +93,27 @@ export class DeploymentsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     public async refreshImpl(): Promise<void> {
         const siteConfig: SiteConfig = await this.root.client.getSiteConfig();
+        const sourceControl: SiteSourceControl = await this.root.client.getSourceControl();
         if (siteConfig.scmType === ScmType.GitHub || siteConfig.scmType === ScmType.LocalGit) {
```

Comment:
I know it's not completely related to the change in this PR, but I just realized your logic for setting contextValue here is different than the logic in the constructor. Which logic is correct? I would prefer to consolidate the logic into one getter as described in the other comment.